src/test/testing_common.c

```diff
@@ -89,6 +89,11 @@ setup_directory(void)
                  (int)getpid(), rnd32);
     r = mkdir(temp_dir);
   }
+#elif defined(__ANDROID__)
+  r = mkdir("/data/local/tmp/tor", 0700);
```

Comment:
We should check this return value.

Reply:
(If different users all run this test, they seem to get subdirectories of the same directory, owned by an earlier user. That seems dangerous.)

Reply:
There are no "different users" in Android in terms of UIDs for file permissions.  Each app gets assigned its own UID.  Apps cannot write to _/data/local/tmp_, only root or `adb shell` connections from a laptop.  So if the same device is connected to a different laptop or user account on that laptop, `adb shell` will still log into the Android emulator as the "shell" user.

Seems pedantic to check that return value, since if it fails, the next `mkdir()` would also fail.  But I'll do it if you want.